app/uk/gov/hmrc/agentaccesscontrol/service/AuthorisationService.scala

```diff
@@ -30,17 +30,20 @@ class AuthorisationService(cesaAuthorisationService: CesaAuthorisationService,
   extends LoggingAuthorisationResults {
 
   def isAuthorised(agentCode: AgentCode, saUtr: SaUtr)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Boolean] =
-    authConnector.currentAgentIdentifiers().flatMap {
-      case (Some(saAgentReference), ggCredentialId) =>
+    authConnector.currentAgentIdentifiers()
+      .flatMap {
+      case Some((Some(saAgentReference), ggCredentialId)) =>
         val results = cesaAuthorisationService.isAuthorisedInCesa(agentCode, saAgentReference, saUtr) zip
           ggAuthorisationService.isAuthorisedInGovernmentGateway(agentCode, ggCredentialId, saUtr)
         results.map { case (cesa, ggw) => {
           val result = cesa && ggw
           logResult(cesa, ggw, agentCode, ggCredentialId, saUtr, result)
           result
         } }
-      case (None, _) =>
+      case Some((None, _)) =>
         Future successful notAuthorised(s"No 6 digit agent reference found for agent $agentCode")
+      case None =>
+        Future successful notAuthorised(s"Agent $agentCode is not logged in")
```

Comment:
It might be clearer to log "No user is logged in".

The current message is not _untrue_ but if I've understood the code correctly then None means "nobody is logged in" not "nobody is logged in OR somebody is logged in but their agent code isn't the expected agent code" which the current message _implies_.